contracts/SnappAuctionChallenge.sol

```diff
@@ -126,13 +122,11 @@ contract SnappAuctionChallenge {
 
     function challengeStateTransition(
         bytes memory pricesAndVolumes,
-        bytes memory orders,
-        bytes memory stateRoots,
+        bytes memory orders,      // Should this be "challengeAuctionSettlement"?
+        bytes memory stateRoots,  // What are these and how many are they?
         bytes memory merklePaths,
-        uint8 stateRootIndex
-    ) public returns (bool) {
-        require(!checkPriceAndVolumeData(pricesAndVolumes), "Wrong prices or volumes");
-        require(!checkOrderData(orders), "Wrong order data");
+        uint8 stateRootIndex      // Is this checkPointIndex? Will we also need stateIndex?
```

Comment:
I don't think we need stateIndex. Yes, this is the checkpoint index that we claim has been incorrectly computed. We therefore take the intermediate stateRoot at that index, apply the orders as defined by the state transition logic (increment buyVolume, deduct sellVolume) and verify that the intermediate state root at `index i+1` is indeed the one claimed by the solution provider.

Reply:
Right, but when the challenge is successful, don't we need to know which state index to revert? Or have the details involving the actual rollback been suppressed here.

Reply:
The rollback logic is orthogonal to the verification logic. I imagine the verifier being a smart contract that is whitelisted in the SnappAuction.sol to trigger a rollback. 
You are right that in order to trigger the rollback we need to know which stateTransition we are challenging, so yes we will require the stateIndex that is being challenged on all of the methods above.